src/spatial_reactionsystem.jl

```diff
@@ -0,0 +1,142 @@
+        prods1_r2 = map(sub -> sym_to_var(sub,rs_combined,e.dst,lattice), first.(sr.products[1]))
+        prods2_r2 = map(sub -> sym_to_var(sub,rs_combined,e.src,lattice), first.(sr.products[2]))
+
+        push!(spatial_reactions,Reaction(sr.rate,[subs1_r1;subs2_r1],[prods1_r1;prods2_r1],[last.(sr.substrates[1]);last.(sr.substrates[2])],[last.(sr.products[1]);last.(sr.products[2])]))
+        push!(spatial_reactions,Reaction(sr.rate,[subs1_r2;subs2_r2],[prods1_r2;prods2_r2],[last.(sr.substrates[1]);last.(sr.substrates[2])],[last.(sr.products[1]);last.(sr.products[2])]))
+    end
+
+    # Adds the spatial reactions to the initial network.
+    return extend(rs_combined,ReactionSystem(spatial_reactions,rs_combined.iv;name=rs_combined.name))
+end
+
+# Helper functions.
+get_name(comp::Int64,lattice::MetaGraph) = has_prop(lattice,comp,:name) ? get_prop(lattice,comp,:name) : Symbol("container___",comp)
+make_subsystem(rs::ReactionSystem,comp::Int64,lattice::MetaGraph) = @set! rs.name = get_name(comp,lattice)
+sym_to_var(sym,rs,comp,lattice) = rs.var_to_name[Symbol(get_name(comp,lattice),:₊,sym)]
+
+### ODEProblem ###
+# Creates an ODEProblem from a LatticeReactionSystem.
+# Currently parameter values are identical across the grid.
+# Default species concentrations are set in the input to the problem. 
+# If a specific default value is given in the input value in the graph, that is used isntead.
+function DiffEqBase.ODEProblem(lrs::LatticeReactionSystem, u0, tspan,
+                               p = DiffEqBase.NullParameters(), args...;
+                               check_length = false, name = nameof(lrs.rs),
+                               combinatoric_ratelaws = get_combinatoric_ratelaws(lrs.rs),
+                               include_zero_odes = true, remove_conserved = false,
+                               checks = false, kwargs...)
+    
+    # Converts the [:X => ...] to similar, but for each compartment variable.
+    state_names = Symbol.(getfield.(states(lrs.rs_base),:f)) # Not sure if there's a better way to do this.
+    ps_names = Symbol.(parameters(lrs.rs_base)) # Not sure if there's a better way to do this.
+    u0_full = vcat(map(comp -> map(species -> sym_to_var(species,lrs.rs,comp,lrs.lattice) => get_val(comp,species,lrs.lattice,u0), state_names), vertices(lrs.lattice))...)
+    p_full = vcat(map(comp -> map(species -> sym_to_var(species,lrs.rs,comp,lrs.lattice) => get_val(comp,species,lrs.lattice,p), ps_names), vertices(lrs.lattice))...)
+
+    u0map = symmap_to_varmap(lrs.rs, u0_full)
+    pmap_lattice = symmap_to_varmap(lrs.rs, p_full)
+    pmap_global = symmap_to_varmap(lrs.rs,filter(p_pair->!in(p_pair[1],ps_names), p))
+
+    osys = convert(ODESystem, lrs.rs; name, combinatoric_ratelaws, include_zero_odes, checks,
+                   remove_conserved)
+    return ODEProblem(osys, u0map, tspan, [pmap_lattice;pmap_global], args...; check_length, kwargs...)
```

Comment:
This is missing the connection to the PDE wrapping system. Please make this make use of the growing PDESolution standards.

Reply:
Yes, for now, I just wanted to make something that works, and then try to integrate it better.

Reply:
Also, since this is just an ODE/SDE/SSA over connected cells, are there ways to utilise PDE solvers more generally?
(in this example it is just a grid, but that will probably be an exception)